### ISSUE TITLE
pip.install_args are not used when spawning cmd in process.lua

### ISSUE THREAD

**Issue by 3699394** [williamboman/mason.nvim#1224]
### I've searched open issues for similar requests

Yes

### I've recently downloaded the latest plugin version of mason.nvim

- [x] Yes

### Problem description

pip.install_args in setup are not used when spawning a process in process.lua

### Expected behavior

pypi installer should respect pip.install_args

### Affected packages

python packages

### Installation log

```Text
[DEBUG Mon 17 Apr 2023 03:11:25 PM HKT] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 12719
[DEBUG Mon 17 Apr 2023 03:11:28 PM HKT] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:148: Job pid=12444 exited with exit_code=1, signal=0
[INFO  Mon 17 Apr 2023 03:11:28 PM HKT] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:166: Executing installer for Package(name=mypy) {}
[ERROR Mon 17 Apr 2023 03:11:28 PM HKT] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:200: Installation failed for Package(name=prettier) error=spawn: npm failed with exit code 1 and signal 0. 
[DEBUG Mon 17 Apr 2023 03:11:28 PM HKT] .../.local/share/nvim/lazy/mason.nvim/lua/mason-core/fs.lua:47: fs: rmrf /home/test/.local/share/nvim/mason/.packages/prettier
[DEBUG Mon 17 Apr 2023 03:11:28 PM HKT] .../.local/share/nvim/lazy/mason.nvim/lua/mason-core/fs.lua:71: fs: mkdirp /home/test/.local/share/nvim/mason/.packages/mypy
[DEBUG Mon 17 Apr 2023 03:11:28 PM HKT] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:39: Building receipt for Package(name=prettier)
[DEBUG Mon 17 Apr 2023 03:11:28 PM HKT] ...zy/mason.nvim/lua/mason-core/installer/registry/init.lua:142: Compiling installer. mypy {}
[DEBUG Mon 17 Apr 2023 03:11:28 PM HKT] ...zy/mason.nvim/lua/mason-core/installer/registry/init.lua:106: Parsing spec mypy {}
[DEBUG Mon 17 Apr 2023 03:11:28 PM HKT] ...zy/mason.nvim/lua/mason-core/installer/managers/pypi.lua:55: pypi: init
[DEBUG Mon 17 Apr 2023 03:11:28 PM HKT] ...vim/lazy/mason.nvim/lua/mason-core/installer/context.lua:212: Promoting cwd "/home/test/.local/share/nvim/mason/.packages/mypy" to "/home/test/.local/share/nvim/mason/packages/mypy"
[DEBUG Mon 17 Apr 2023 03:11:28 PM HKT] .../.local/share/nvim/lazy/mason.nvim/lua/mason-core/fs.lua:65: fs: mkdir /home/test/.local/share/nvim/mason/packages/mypy
[DEBUG Mon 17 Apr 2023 03:11:28 PM HKT] .../.local/share/nvim/lazy/mason.nvim/lua/mason-core/fs.lua:84: fs: rename /home/test/.local/share/nvim/mason/.packages/mypy /home/test/.local/share/nvim/mason/packages/mypy
[DEBUG Mon 17 Apr 2023 03:11:28 PM HKT] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="python3", spawn_opts={
  args = { "-m", "venv", "venv" },
  cwd = "/home/test/.local/share/nvim/mason/packages/mypy"
}
[DEBUG Mon 17 Apr 2023 03:11:28 PM HKT] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 13072
[DEBUG Mon 17 Apr 2023 03:11:29 PM HKT] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:148: Job pid=12166 exited with exit_code=0, signal=0
[DEBUG Mon 17 Apr 2023 03:11:29 PM HKT] ...zy/mason.nvim/lua/mason-core/installer/managers/pypi.lua:85: pypi: install "black" "23.3.0"
[DEBUG Mon 17 Apr 2023 03:11:29 PM HKT] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="/home/test/.local/share/nvim/mason/packages/black/venv/bin/python", spawn_opts={
  args = { "-m", "pip", "--disable-pip-version-check", "install", "-U", "black==23.3.0" },
  cwd = "/home/test/.local/share/nvim/mason/packages/black"
}
```

### Neovim version (>= 0.7)

NVIM v0.10.0-dev-102+g85bc9e897
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3

### Operating system/version

Linux 3.10.0-1160.76.1.el7.x86_64 #1 SMP Tue Jul 26 14:15:37 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Healthcheck

```Text
mason: require("mason.health").check()
  3
  4 mason.nvim
  5 - OK neovim version >= 0.7.0
  6
  7 mason.nvim [Registries]
  8 - OK Registry require("mason-registry.index") is installed.
  9 - OK Registry github.com/mason-org/mason-registry version: 2023-04-17-shady-john is installed.
 10
 11 mason.nvim [Core utils]
 12 - OK unzip: UnZip 6.00 of 20 April 2009, by Info-ZIP.  Maintained by C. Spieler.  Send
 13 - OK wget: GNU Wget 1.14 built on linux-gnu.
 14 - OK curl: curl 7.29.0 (x86_64-redhat-linux-gnu) libcurl/7.29.0 NSS/3.44 zlib/1.2.3 libidn/1.29 libssh2/1.8.0
 15 - OK gzip: gzip 1.5
 16 - OK tar: tar (GNU tar) 1.26
 17 - WARNING pwsh: not available
 18   - ADVICE:
 19     - spawn: pwsh failed with exit code - and signal -. pwsh is not executable
 20 - OK bash: GNU bash, version 4.2.46(2)-release (x86_64-redhat-linux-gnu)
 21 - OK sh: Ok
 22
 23 mason.nvim [Languages]
 24 - WARNING luarocks: not available
 25   - ADVICE:
 26     - spawn: luarocks failed with exit code - and signal -. luarocks is not executable
 27 - WARNING Composer: not available
 28   - ADVICE:
 29     - spawn: composer failed with exit code - and signal -. composer is not executable
 30 - WARNING PHP: not available
 31   - ADVICE:
 32     - spawn: php failed with exit code - and signal -. php is not executable
 33 - WARNING RubyGem: not available
 34   - ADVICE:
 35     - spawn: gem failed with exit code - and signal -. Could not find executable "gem" in path.
 36
 37 - OK cargo: cargo 1.68.0 (115f34552 2023-02-26)
 38 - OK Go: go version go1.19.3 linux/amd64
 39 - WARNING node: unsupported version v8.11.1
 40   - ADVICE:
 41     - Node version must be >= 14
 42 - WARNING julia: not available
 43   - ADVICE:
 44     - spawn: julia failed with exit code - and signal -. julia is not executable
 45 - WARNING npm: unsupported version 5.6.0
 46   - ADVICE:
 47     - npm version must be >= 6
 48 - OK python3: Python 3.8.0
 49 - OK Ruby: ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-linux]
 50 - OK pip3: pip 22.2.2 from /home/test/.local/lib/python3.8/site-packages/pip (python 3.8)
 51 - OK java: java version "14.0.1" 2020-04-14
 52 - OK javac: javac 14.0.1
 53
 54 mason.nvim [GitHub]
 55 - OK GitHub API rate limit. Used: 13. Remaining: 47. Limit: 60. Reset: Mon 17 Apr 2023 04:36:47 PM HKT.
 56   Install and authenticate via gh-cli to increase rate limit.
```

**Comment by williamboman** [comment]
How do you set up mason, how does the config table look?

**Comment by 3699394** [comment]
`{
                "williamboman/mason.nvim",
                dependencies = { "williamboman/mason-lspconfig.nvim" },
                config = function()
                        require("mason").setup({
                                log_level = vim.log.levels.DEBUG,
                                pip = {
                                        install_args = {"--trusted-host", "pypi.org", "--trusted-host", "pypi.python.org", "--trusted-host", "files.pythonhosted.org"}
                                }
                        })
                end,
        },`

setting up with lazy.nvim

**Comment by williamboman** [comment]
Should be fixed in #1263!
